python/pyspark/ml/torch/distributor.py

```diff
@@ -643,13 +664,52 @@ def _setup_files(train_fn: Callable, *args: Any) -> Generator[Tuple[str, str], N
         finally:
             TorchDistributor._cleanup_files(save_dir)
 
+    @staticmethod
+    @contextmanager
+    def _setup_spark_partition_data(partition_data_iterator, input_schema_json):
+        from pyspark.sql.pandas.serializers import ArrowStreamSerializer
+        from pyspark.files import SparkFiles
+        import json
+
+        if input_schema_json is None:
+            yield
+            return
+
+        # We need to temporarily write partition data into a temp dir,
+        # partition data might be huge, so we need to write it under
+        # configured `SPARK_LOCAL_DIRS`.
+        save_dir = TorchDistributor._create_save_dir(root_dir=SparkFiles.getRootDirectory())
+
+        try:
+            serializer = ArrowStreamSerializer()
+            arrow_file_path = os.path.join(save_dir, "data.arrow")
```

Comment:
You might need to remove if the directory already exists. If Python UDF fails (with Python worker reused), it will retries to use this path if I am not mistaken.

Reply:
Oh, actually NVM. `with open(arrow_file_path, "wb") as f:` will write it from the first.